driver/main.c

```diff
@@ -1778,6 +1794,12 @@ static int record_event_consumer(struct ppm_consumer_t *consumer,
 		{
 			return res;
 		}
+
+		if (event_type & 1 && consumer->drop_failed)
+		{
+			retval = (int64_t)syscall_get_return_value(current, event_datap->event_info.syscall_data.regs);
```

Comment:
do we miss a check to `retval<0`? it seems like we are dropping all exit events in this way

Reply:
Moreover, why do we check it here and not in the `syscall_exit_probe` dedicated code? we should have the ret value directly from the tracepoint

Reply:
It's just that we are doing all the dropping checks there, therefore i added it there for coherence.

Reply:
Yep, for kmod there is also the problem that we need a consumer to work on; we cannot stay in `sys_exit`.
But we've got `tp_type == SYS_EXIT` and we've got multiple other checks like: `PPMC_SYSCALL`

Reply:
uh that's true! good point! damn multi-consumer logic!